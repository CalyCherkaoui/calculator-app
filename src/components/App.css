@import url('https://fonts.googleapis.com/css2?family=Raleway&family=Ultra&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Abril+Fatface&display=swap');

:root {
  --my_nav_backg: rgba(253, 253, 253, 2);
  --my_dark_main: #1b2d38;
  --my_light_second: #2EB641;
  --tangerine: #da4221;
  --my-light-blue: #2ab2c0;
  --text-dark: #272727;
  --classic-blue: #0f4c81;
}

*,
*::before,
*::after {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

body {
  font-family: "Roboto", sans-serif;
  min-height: 100vh;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

#root {
  width: 100%;
  height: 100vh;
}

/* navigation bar */

.navbar_wrapper {
  width: 100%;
  margin: 0;
  height: 5rem;
  padding: 0.3rem 0.1rem 0.8rem 0.1rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: var(--my_nav_backg);

  -webkit-box-shadow: 0 2px 6px 0 rgba(10, 22, 78, 0.15);
  -moz-box-shadow: 0 2px 6px 0 rgba(10, 22, 78, 0.15);
  box-shadow: 0 2px 6px 0 rgba(10, 22, 78, 0.15);
}

.logo {
  margin-left: 1rem;
  font-family: 'Ultra', serif, Arial;
  font-size: 1.8rem;
  color: #14213d;
  line-height: 2;
}

.logo span {
  font-family: 'Ultra', serif, Arial;
  font-size: 1.8rem;
  color: #fca311;
  line-height: 2;
}

.nav_list {
  list-style: none;
  margin: 0;
  display: flex;
  flex-direction: row;
  margin: 1rem;
  /* background-color: blue; */
}

.nav_item {
  margin: 1rem;
}

.nav_link {
  text-decoration: none;
  padding: 0.2rem 0.6rem;
  background-color: transparent;
  border: none;
  cursor: pointer;
  outline: none;
  transition: 0.3s;
  text-transform: uppercase;
  font-size: 1.2rem;
  font-family: 'Raleway', sans-serif;
  font-weight: 900;
  color: #03045e;
}

.nav_link:hover {
  border-bottom: 2px solid #fca311;
  color: #fca311;
}

.displayer {
  width: 100%;
  height: 8rem;
  border: brown solid 3px;
}

.buttonsPanel {
  border: chartreuse solid 3px;
}

.buttons_row {
  display: flex;
  flex-direction: row;
}

.btn {
  flex-grow: 1;
}

button {
  padding: 10px;
}
