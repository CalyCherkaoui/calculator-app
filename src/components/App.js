// import React, { Component } from 'react';
import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from './logic/calculate';
import './App.css';

// class App extends Component {
//   constructor(props) {
//     super(props);
//     /* eslint-disable react/no-unused-state */
//     this.state = {
//       total: '0',
//       next: '',
//       operation: '',
//     };

//     this.handleClick = this.handleClick.bind(this);
//   }

//   handleClick(btn) {
//     this.setState(
//       (state) => {
//         const data = { total: state.total, next: state.next, operation: state.operation };
//         return calculate(data, btn);
//       },
//     );
//   }

//   render() {
//     return (
//       <>
//         <Display result={this.state} />
//         <ButtonPanel clickHandler={this.handleClick} />
//       </>
//     );
//   }
// }

const App = () => {
  const [state, setState] = useState({ total: '', next: '', operation: '' });

  const handleClick = btn => {
    setState(calculate(state, btn));
  }
  return <div>app</div>;
};

export default App;
