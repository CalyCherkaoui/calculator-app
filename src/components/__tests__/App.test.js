import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../App';

describe(
  'App component',
  () => {
    it('Matches App snapshot', () => {
      const comp = renderer.create(<BrowserRouter><App /></BrowserRouter>);
      expect(comp).toMatchSnapshot();
    });
  },
);
