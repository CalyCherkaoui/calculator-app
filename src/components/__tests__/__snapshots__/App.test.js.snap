// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Matches App snapshot 1`] = `
Array [
  <div
    className="navbar_wrapper"
>
    <div
        className="logo"
    >
        Math
        <span>
             Magicians
        </span>
    </div>
    <ul
        className="nav_list"
    >
        <li
            className="nav_item"
        >
            <a
                className="nav_link"
                href="/"
                onClick={[Function]}
            >
                Home
            </a>
        </li>
        <li
            className="nav_item"
        >
            <a
                className="nav_link"
                href="/Quote"
                onClick={[Function]}
            >
                Quote
            </a>
        </li>
        <li
            className="nav_item"
        >
            <a
                className="nav_link"
                href="/Calculator"
                onClick={[Function]}
            >
                Calculator
            </a>
        </li>
    </ul>
</div>,
  <div
    className="home"
>
    <h1
        className="home_title"
    >
        Welcome to the magic Calculator
    </h1>
    <p
        className="home_text"
    >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
</div>,
]
`;
